### PR TITLE
Add cmdline completion for :e ++opt=arg and :terminal [++options]

### PR DESCRIPTION
**Pull request by ychin**

### CODE REVIEW

**Comment by ychin** [comment]
Just more cmdline completion for some of the other common pain points I encounter.

++opt

terminal options

File: src/testdir/test_cmdline.vim [hunk 1]
```diff
@@ -1083,6 +1083,47 @@ func Test_cmdline_complete_expression()
...
+  call assert_equal('mac', getcompletion('args ++fileformat=m', 'cmdline')[0])
+  call assert_equal('utf-8', getcompletion('split ++enc=ut*-8', 'cmdline')[0])
+  call assert_equal('latin1', getcompletion('tabedit ++encoding=lati', 'cmdline')[0])
+  call assert_equal('keep', getcompletion('edit ++bad=k', 'cmdline')[0])
+
+  call assert_equal([], getcompletion('edit ++bogus=', 'cmdline'))
+
+  " completion should skip the ++opt and continue
+  defer delete('Xaaaaa.txt')
+  call writefile([], 'Xaaaaa.txt')
```
Comment (zeertzjq) [inline src/testdir/test_cmdline.vim]:
```suggestion
  call writefile([], 'Xaaaaa.txt', 'D')
```

Comment (ychin) [reply src/testdir/test_cmdline.vim]: Fixed

**Comment by chrisbra** [comment]
oh very nice. thanks.

**Pull request closed without merging**